### ISSUE TITLE
Some componet names break dex

### ISSUE THREAD

**Issue by halatmit** [mit-cml/appinventor-sources#514]
The attached project has a label whose name is LBL_​2214598160'
The project builds (apparently) but when you got to install it you get an INSTALL FAILED DEXOPT error.
An look at the log shows:

E/dex2oat (28607): Failed to open dex from file descriptor for zip file '/data/app/appinventor.ai_halatmitdotedu.badlbl-1/base.apk': Failure to verify dex file '/data/app/appinventor.ai_halatmitdotedu.badlbl-1/base.apk': Invalid field name: 'LBL_​2214598160'
E/installd(  196): DexInv: --- END '/data/app/appinventor.ai_halatmitdotedu.badlbl-1/base.apk' --- status=0x0100, process failed

I don't know what dex doesn't like about that name.

If we figure it out, we should put in a check in App Inventor to reject the bad names.

Here are the aia and the apk

https://drive.google.com/open?id=0B3jsksMcCW5bUDQ4YVllWGR6LXM
https://drive.google.com/open?id=0B3jsksMcCW5bVkIyMjdlNTNIYnc

**Comment by chenglonkhin** [comment]
@halatmit Do you still have the aia and apk? The Google Drive links lead to a Not Found.

**Comment by halatmit** [comment]
No, I don't have these.   Try building an app that have a label named LBL_​2214598160. 
If that works, then make a comment here saying that it could not be duplicated and close the issue

**Comment by chenglonkhin** [comment]
Just tried the application. Like you said, the application builds but fails to install on the phone. I'll see if I can determine what kinds of names triggers this.

**Comment by ewpatton** [comment]
Poked around this a bit and here's a record for later:

1. Created an APK with the bad label name per the original issue.
2. Attempt to install on phone, confirmed that the app does not install with the error given.
3. Looked in the sources for Android OSP and found that everything in the field name ought to pass the isValidMethodName/isValidMethodNameUtf8 check (either in the art or libdex modules).
4. Attempted to dexdump the APK to inspect it, but this fails with the same error.
5. Extracted classes.dex and examined it with `xxd`.
6. Hex analysis reveals that there are extra bytes trailing the LBL_ portion of the name and preceding the numbers.
7. As an admin, I generated the YAIL and exported the project. Examination of the YAIL indicates that the byte sequence (which is not valid UTF-8) exists before compilation.
8. Examined the SCM file. This character is present there and is shown as \u200b.
9. Decoding \u200b reveals that this is UTF character "ZERO WIDTH SPACE", which is presumably not a valid character for a Java identifier.

Since whitespace is not allowed in valid Java identifiers, I think we should do the following:
- [ ] Update `TextValidators.isValidComponentIdentifier` to include a check for whitespace.
- [ ] Update the YAIL generator to replace whitespace with something like underscore or remove the whitespace entirely. If such a component is found to exist and a collision occurs by removing the whitespace, prompt the user to rename it.
